domain/machine/service/service_test.go

```diff
@@ -90,3 +90,66 @@ func (s *serviceSuite) TestGetLifeError(c *gc.C) {
 	c.Check(err, jc.ErrorIs, rErr)
 	c.Assert(err, gc.ErrorMatches, `getting life status for machine "666": boom`)
 }
+
+func (s *serviceSuite) TestListAllMachinesSuccess(c *gc.C) {
+	defer s.setupMocks(c).Finish()
+
+	s.state.EXPECT().ListAllMachines(gomock.Any()).Return([]string{"666"}, nil)
+
+	machines, err := NewService(s.state).ListAllMachines(context.Background())
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(machines, gc.DeepEquals, []string{"666"})
+}
+
+func (s *serviceSuite) TestListAllMachinesError(c *gc.C) {
+	defer s.setupMocks(c).Finish()
+
+	rErr := errors.New("boom")
+	s.state.EXPECT().ListAllMachines(gomock.Any()).Return(nil, rErr)
+
+	machines, err := NewService(s.state).ListAllMachines(context.Background())
+	c.Check(err, jc.ErrorIs, rErr)
+	c.Assert(machines, gc.IsNil)
+}
+
+func (s *serviceSuite) TestInstanceIdSuccess(c *gc.C) {
+	defer s.setupMocks(c).Finish()
+
+	s.state.EXPECT().InstanceId(gomock.Any(), "666").Return("123", nil)
+
+	instanceId, err := NewService(s.state).InstanceId(context.Background(), "666")
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(instanceId, gc.Equals, "123")
+}
+
+func (s *serviceSuite) TestInstanceIdError(c *gc.C) {
+	defer s.setupMocks(c).Finish()
+
+	rErr := errors.New("boom")
+	s.state.EXPECT().InstanceId(gomock.Any(), "666").Return("", rErr)
+
+	instanceId, err := NewService(s.state).InstanceId(context.Background(), "666")
+	c.Check(err, jc.ErrorIs, rErr)
+	c.Assert(instanceId, gc.Equals, "")
+}
+
+func (s *serviceSuite) TestInstanceStatusSuccess(c *gc.C) {
+	defer s.setupMocks(c).Finish()
+
+	s.state.EXPECT().InstanceStatus(gomock.Any(), "666").Return("running", nil)
+
+	instanceStatus, err := NewService(s.state).InstanceStatus(context.Background(), "666")
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(instanceStatus, gc.Equals, "running")
+}
+
+func (s *serviceSuite) TestInstanceStatusError(c *gc.C) {
```

Comment:
Same comment as above re docs for test. We care that the state layer error is being preserved. This is what the test is doing and we should be explicit about.